File: src/api/schemas/audit_log.py
"""Pydantic schemas for audit log responses."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class Pagination(BaseModel):
    page: int
    page_size: int
    total: int
    total_pages: int


class AuditLogResponse(BaseModel):
    id: int
    user_id: str | None = None
    username: str | None = None
    org_id: str | None = None
    team_id: str | None = None
    action: str
    resource_type: str
    resource_id: str | None = None
    ip_address: str | None = None
    user_agent: str | None = None
    request_method: str | None = None
    request_path: str | None = None
    status_code: int | None = None
    duration_ms: int | None = None
    detail: dict | None = None
    created_at: datetime


class AuditLogListResponse(BaseModel):
    data: list[AuditLogResponse]
    pagination: Pagination

File: web/src/types/auditLog.ts
import type { PaginatedResponse } from './api';

export interface AuditLog {
  id: number;
  user_id?: string;
  username?: string;
  org_id?: string;
  team_id?: string;
  action: string;
  resource_type: string;
  resource_id?: string;
  ip_address?: string;
  user_agent?: string;
  request_method?: string;
  request_path?: string;
  status_code?: number;
  duration_ms?: number;
  detail?: Record<string, unknown>;
  created_at: string;
}

export type AuditLogListResponse = PaginatedResponse<AuditLog>;

export interface AuditLogQueryParams {
  page?: number;
  page_size?: number;
  user_id?: string;
  action?: string;
  resource_type?: string;
  start_date?: string;
  end_date?: string;
}
